Added toggle button

lj894 opened the pull request:
Added a toggle button on the UI to allow user toggle the status of a given record.

lj894 committed:
Added toggle button

Added a toggle button on the UI to allow user toggle the status of a given record.

sushanthsamala commented on service/static/js/rest_api.js:
```diff
@@ -271,4 +271,33 @@ $(function () {
+            })
+
+            ajax.done(function(res){
+                flash_message("Success")
```
We need to update the form with the updated recommendation don't we?

lj894 replied:
Not sure. Are you saying if after I used toggle button the form should refill with the new data? 
In this regard I think a better way would be we clear the form and list the new data in the bottom table. So it's more intuitive that an update or a toggle has done.

lj894 replied:
Also it looks like the toggle endpoint only returns the status field on success. So either we need to call a get afterwards to get the whole data, or we probably need another story to modify the return object of toggle endpoint.

sushanthsamala replied:
Yeah sure we can have the table at the bottom updated as well, but there is no point in having out dated data in the form after toggle has been executed. Also, since this is a very minor I think it is okay to do it in the same story, but you are welcome to create another story.

lj894 replied:
Okay I will clean it up.

sushanthsamala replied:
Hey @lj894, could you also please add $("#recommendation_status").val("") in line 23 clean form data function, this will make it possible to search all recommendations including false and true?

lj894 committed:
Fill out form after toggle

Trigger search after user call toggle api.

sushanthsamala commented on service/static/js/rest_api.js:
```diff
@@ -271,4 +271,34 @@ $(function () {
+
+            ajax.done(function(res){
+                flash_message("Success");
+                $("#search-btn").trigger('click');
```
Hi @lj894, you could just do 'update_form_data(res)', I think triggering the search button will cause a delay as a new request has to be sent and received.

lj894 replied:
Ok. Because the toggle only returns the status, not the other attributes. so I thought it'd be safe to do another GET to make sure we get all the information from API so the toggle won't accidently update anything else. But I guess since we have tested the API elsewhere we can safe to assume the API works as we expected.

sushanthsamala replied:
um okay, I thought we agreed on changing the toggle endpoint to return the entire recommendation, not just the status, is there any reason to return only the status?

lj894 committed:
replace search with update_form_data

replace search with update_form_data

lj894 committed:
update toggle api to return the full class

Update toggle api to return the full class.
Clear button should also clear the ststus field.

lj894 committed:
updated bdd test

updated bdd test.
